shell/src/main/java/alluxio/cli/Check.java

```diff
@@ -0,0 +1,238 @@
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.TreeMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utility for checking Alluxio environment.
+ */
+public final class Check {
+  private static final String USAGE = "USAGE: Check [task full name or prefix]\n\n"
+      + "Check environment for Alluxio";
+
+  private static final Options OPTIONS = new Options();
+
+  private static final Map<String, CheckTask> TASK_MAP = initializeTaskMap();
+
+  private static Map<String, CheckTask> initializeTaskMap() {
+    Map<String, CheckTask> taskMap = new TreeMap<>();
+    taskMap.put("master.port.available", new MasterRpcCheckTask());
+    taskMap.put("master.workers.ssh.reachable", new MasterWorkersSshCheckTask());
+    taskMap.put("worker.port.available", new WorkerRpcCheckTask());
+    return taskMap;
+  }
+
+  private interface CheckTask {
+    boolean check();
+  }
+
+  private static boolean isLocalPortAvailable(int port) {
+    try (ServerSocket ss = new ServerSocket(port)) {
+      ss.setReuseAddress(true);
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAddressReachable(String hostname, int port) {
+    try (Socket socket = new Socket(hostname, port)) {
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  private static boolean isAlluxioRunning(String className) {
+    String[] command = {"bash", "-c",
+                        "ps -Aww -o command | grep -i \"[j]ava\" | grep " + className};
+    try {
+      Process p = Runtime.getRuntime().exec(command);
+      try (InputStreamReader input = new InputStreamReader(p.getInputStream())) {
+        if (input.read() >= 0) {
+          return true;
+        }
+      }
+      return false;
+    } catch (IOException e) {
+      System.err.format("Unable to check Alluxio status: %s.%n", e.getMessage());
+      return false;
+    }
+  }
+
+  private static class MasterRpcCheckTask implements CheckTask {
+
+    public MasterRpcCheckTask() {}
+
+    @Override
+    public boolean check() {
+      if (isAlluxioRunning("alluxio.master.AlluxioMaster")) {
+        System.out.println("Alluxio master is already running. Skip checking.");
+        return true;
+      }
+      int port = Configuration.getInt(PropertyKey.MASTER_RPC_PORT);
+      if (!isLocalPortAvailable(port)) {
+        System.err.format("Master RPC port %d is not available.%n", port);
+        return false;
+      }
+      return true;
+    }
+  }
+
+  private static class WorkerRpcCheckTask implements CheckTask {
+
+    public WorkerRpcCheckTask() {}
+
+    @Override
+    public boolean check() {
+      if (isAlluxioRunning("alluxio.worker.AlluxioWorker")) {
```

Comment:
Alluxio is running in a cluster, and this command may run from any node. However, this assumes all the master and worker are on the current node. I think an invocation of this check on remote nodes is needed, if discovering remote port availability is not possible from the current node.